src/rules/partial-no-import/__tests__/index.js

```diff
@@ -1,319 +1,204 @@
-    .then(result => {
-      const warnings = result.warnings();
-
-      expect(warnings).toHaveLength(0);
-      done();
-    });
-});
-
-test("Multiple imports in a partial.", done => {
-  expect.assertions(1);
-  postcss([rule()])
-    .process(
-      '@import "_bootstrap/variables";\n@import "_font-awesome/variables";',
-      {
-        from: path.join(__dirname, "_variables.scss")
-      }
-    )
-    .then(result => {
-      const warnings = result.warnings();
-
-      expect(warnings).toHaveLength(2);
-      done();
-    });
-});
-
-test("Import from a non-partial SCSS-file.", done => {
-  expect.assertions(1);
-  postcss([rule()])
-    .process(
-      '@import "bootstrap/variables";\n@import "font-awesome/variables";',
-      {
-        from: path.join(__dirname, "_der", "variables.scss")
-      }
-    )
-    .then(result => {
-      const warnings = result.warnings();
-
-      expect(warnings).toHaveLength(0);
-      done();
-    });
+const { messages, ruleName } = rule;
+
+testRule({
+  ruleName,
+  config: [true],
+
+  accept: [
+    {
+      description: "Import a file from non-partial .scss",
+      code: "@import 'file.scss';",
+      codeFilename: path.join(__dirname, "test.scss")
+    },
+    {
+      description: "Ignores empty imports (Sass will throw an error instead)",
+      code: "@import '';",
+      codeFilename: path.join(__dirname, "_test.scss")
+    },
+    {
+      description: "Ignores empty imports (Sass will throw an error instead) 2",
+      code: '@import " ";',
+      codeFilename: path.join(__dirname, "_test.scss")
+    },
+    {
+      description: "Import a file from CSS",
+      code: "@import 'file.scss';",
+      codeFilename: path.join(__dirname, "_test.css")
+    },
+    {
+      description: "Import a CSS from a partial .scss",
+      code: "@import 'file.css';",
+      codeFilename: path.join(__dirname, "_test.scss")
+    },
+    {
+      description: "Import a CSS (url) from a partial .scss",
+      code: "@import url('file.css');",
+      codeFilename: path.join(__dirname, "_test.scss")
+    },
+    {
+      description: "Import a CSS (with protocol) from a partial .scss",
+      code: "@import '//file.css';",
+      codeFilename: path.join(__dirname, "_test.scss")
+    },
+    {
+      description:
+        "Import a CSS file (font URL with https) from a partial .scss",
+      code: "@import 'https://fonts.googleapis.com/css?family=Quicksand:300,400,500,700';",
+      codeFilename: path.join(__dirname, "_test.scss")
+    },
+    {
+      description: "Import a CSS (with media) from a partial .scss",
+      code: "@import 'file.scss' screen",
+      codeFilename: path.join(__dirname, "_test.scss")
+    },
+    {
+      description: "Import from a non-partial SCSS-file.",
+      code: '@import "bootstrap/variables";\n@import "font-awesome/variables";',
+      codeFilename: path.join(__dirname, "_der", "variables.scss")
+    }
+  ],
+
+  reject: [
+    {
+      description: "No file specified",
```

Comment:
[note] Each `description` in all the `accept`/`reject` cases is associated with the old description in `test()`. For example, see https://github.com/stylelint-scss/stylelint-scss/pull/1019/files#diff-52a0082c93d02f6ded3b526b7cc8d60de78eda7eabe80bc7f8bb9d85b57e3cc8L12